cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetsModules.java

```diff
@@ -119,7 +119,7 @@ protected void configure() {
 
   @Override
   public Module getDistributedModules() {
-    return getDistributedModules(false);
+    return getDistributedModules(true);
```

Comment:
E.g. If the cdap setting doesn't contain Kafka information for the metadata to publish to, then the binding will be the no-op one, otherwise it is the Kafka one. You can have the binding looks like this:

```java
bind(MetadataChangePublisher.class).toProvider(MetadataChangePublisherProvider.class);

private static final class MetadataChangePublisherProvider implements Provider<MetadataChangePublisher> {
  private final Injector injector;
  private final CConfiguration cConf;

  @Inject
  public MetadataChangePublisherProvider(Injector injector, CConfiguration cConf) {
    ....
  }

  @Override
  public MetadataChangePublisher get() {
    if (cConf.get("some meta data kafka thing")) {
      return injector.getInstance(KafkaMetadataChangePublisher.class);
    }
    return injector.getInstance(NoOpMetadataChangePublisher.class);
  }
}